core/ir/instr_shared.c

```diff
@@ -470,6 +470,25 @@ instr_get_category(instr_t *instr)
 /* in rest of file, directly de-reference for performance (PR 622253) */
 #define instr_get_category inlined_instr_get_category
 
+const char *
+instr_get_category_name(dr_instr_category_t category)
```

Comment:
We should have at least a sanity test of a new API in the DR core's tests (clients/drcachesim is considered separate from the "core").  Please add some spot-checks of category strings, maybe in test_categories() in suite/tests/api/drdecode_x86.c.

Reply:
Done. I added the check in the CATEGORY_CHECK macro, but if you want me to separate the two and add more "spot-checks" for more instructions I can do that.

Reply:
Maybe add one instance of a multi-category instruction?

Reply:
Done. Added one for "uncategorized" as well, since it's kind of an edge case being all 0s.